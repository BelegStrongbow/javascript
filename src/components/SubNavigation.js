import PropTypes from "prop-types";
import React from "react";
import styled from "styled-components";
import colors from "yoast-components/style-guide/colors.json";
import { NavLink, Route } from "react-router-dom";

const activeStyle = "activeSubMenu";

const SubNavigationContainer = styled.ul`
	width: 100%;
	display: flex;
	justify-content: center;
	align-items: center;
	text-align: center;
	margin: auto;
	list-style-type: none;
	margin-bottom: 2em;
	padding: 0;
	border-bottom: 4px solid ${ colors.$color_grey_medium };
`;

const SubNavigationEntry = styled.li`
	min-width: 200px;
	font-size: 1.2em;

	@media screen and ( max-width: 600px ) {
		width: 150px;
		min-width: 0px;
	}
`;

const LinkItem = styled( NavLink )`
	display: block;
	text-decoration: none;
	padding: 0.5em 0;
	color: ${ colors.$color_grey_dark };

	margin-bottom: -4px;
	border-bottom: 4px solid transparent;

	transition: border 200ms ease-out;

	&:hover,
	&:focus,
	&.${ activeStyle } {
		border-bottom-color: ${ colors.$color_pink_dark };
		border-bottom-style: double;
	}

	&.${ activeStyle } {
		color: ${ colors.$color_pink_dark };
		border-bottom-style: solid;
	}
`;

/**
 * The SubNavigation component.
 *
 * @param {Object} props The props to use.
 *
 * @returns {ReactElement} The rendered SubNavigation component.
 */
export default function SubNavigation( props ) {
	return (
		<SubNavigationContainer role="list">
					{ props.itemRoutes.map( function( item ) {
						let isActive = item.isActive || ( match => match );

						return <SubNavigationEntry key={ item.title }>
								<LinkItem activeClassName={ activeStyle } to={ item.path } isActive={ isActive }> { item.title } </LinkItem>
							</SubNavigationEntry>;
					}
					) }
		</SubNavigationContainer>
	);
}

SubNavigation.propTypes = {
	itemRoutes: PropTypes.array.isRequired,
};

/**
 * The SubNavigationItem routes.
 *
 * @param {Object} props The props to use.
 * @returns {ReactElement} The rendered route.
 */
export function SubNavigationItem( props ) {
	return (
		<div>
			{ props.itemRoutes.map( function( route, routeKey ) {
				return <Route exact={true} key={ routeKey } path={ route.path } component={ route.component }/>;
			}
			) }
		</div>
	);
}

SubNavigationItem.propTypes = {
	itemRoutes: PropTypes.array.isRequired,
};
