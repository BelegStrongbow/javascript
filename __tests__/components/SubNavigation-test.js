import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';
import SubNavigation, { SubNavigationItem } from "../../src/components/SubNavigation";

let itemRoutes = [
	{
		component: ( props ) => { return null; },
		path: "/account/subscriptions",
		title: "Subscriptions",
	},
	{
		component: ( props ) => { return null; },
		path: "/account/orders",
		title: "Orders",
	},
	{
		component: ( props ) => { return null; },
		path: "/account/profile",
		title: "Profile",
	},
];

test('the subnavigation component matches the snapshot', () => {
	const component = renderer.create(
		<Router>
			<div>
				<SubNavigation itemRoutes={ itemRoutes }/>
				<SubNavigationItem itemRoutes={ itemRoutes } />
			</div>
		</Router>
		);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
