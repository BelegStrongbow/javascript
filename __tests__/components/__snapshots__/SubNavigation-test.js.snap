exports[`test the subnavigation component matches the snapshot 1`] = `
<div>
  <ul
    className="fXEqiq"
    role="list">
    <li
      className="isgbHI">
      <a
        className="hvvVvi"
        href="/account/subscriptions"
        onClick={[Function]}
        style={undefined}>
         
        Subscriptions
         
      </a>
    </li>
    <li
      className="isgbHI">
      <a
        className="hvvVvi"
        href="/account/orders"
        onClick={[Function]}
        style={undefined}>
         
        Orders
         
      </a>
    </li>
    <li
      className="isgbHI">
      <a
        className="hvvVvi"
        href="/account/profile"
        onClick={[Function]}
        style={undefined}>
         
        Profile
         
      </a>
    </li>
  </ul>
  <div />
</div>
`;
